packages/microsoft_dhcp/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml

```
@@ -373,6 +373,27 @@ processors:
         }
         def hm = new HashMap(params[ctx.event.code]);
         hm.forEach((k, v) -> ctx.event[k] = v);
+  - script:
+      description: Set the QResult description based on numerical value.
+      lang: painless
+      tag: Add QResult Description
+      params:
+        "0":
+          result_description: NoQuarantine
+        "1":
+          result_description: Quarantine
+        "2":
+          result_description: Drop Packet
+        "3":
+          result_description: Probation
+        "6":
+          result_description: No Quarantine Information
+      source: |-
+        if (ctx.microsoft?.dhcp?.result == null || params.get(ctx.microsoft.dhcp.result) == null) {
+          return;
+        }
+        def hm = new HashMap(params[ctx.microsoft.dhcp.result]);
+        hm.forEach((k, v) -> ctx.microsoft.dhcp[k] = v);
```

Do we need an iterator here?
```suggestion
  - script:
      description: Set the QResult description based on numerical value.
      lang: painless
      tag: add_qresult_description
      params:
        "0": NoQuarantine
        "1": Quarantine
        "2": Drop Packet
        "3": Probation
        "6": No Quarantine Information
      source: |-
        if (ctx.microsoft?.dhcp?.result == null) {
          return;
        }
        def desc = params.get(ctx.microsoft.dhcp.result);
        if (desc == null) {
          return;
        }
        ctx.microsoft.dhcp['result_description'] = desc; 
```

---

I went ahead and committed. I have not seen this before. I was following along the way I did the last PR.